semanticdb/metacp/src/main/scala/scala/meta/internal/javacp/Javacp.scala

```diff
@@ -368,22 +367,32 @@ object Javacp {
         case _ => i -= 1
       }
     }
-    throw new IllegalArgumentException(s"Missing $$ or / from symbol '$asmName'")
+    asmName
   }
 
-  private def ssym(asmName: String): String =
-    asmName.replace('$', '#').replace('/', '.') + "#"
+  private def ssym(asmName: String): String = {
+    var fqName = asmName.replace('$', '#').replace('/', '.')
+    locally {
```

Comment:
why locally?

Reply:
Can we at least guard against `if (fqName.endsWith("package-info") || fqName.endsWith("#."))`?

Sure, it's premature optimization but

* this method is VERY hot
* `replace(String, String)` is not cheap

I already felt bad enough doing to `replace(Char, Char)` twice instead of rolling a while loop 😅

Reply:
I'll just reimplement this method as a while loop.